org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/ResourceForIEditorInputFactory.java

```diff
@@ -75,6 +75,11 @@ public Resource createResource(IEditorInput editorInput) {
 		try {
 			if (editorInput instanceof IStorageEditorInput) {
 				IStorage storage = ((IStorageEditorInput) editorInput).getStorage();
+				if (storage.getFullPath() == null) {
+					IFile file = editorInput.getAdapter(IFile.class);
```

Comment:
Should probably be `Adapters.getAdapter(..)` to honor registered adapter factories.

Reply:
Yes, I changed it.